json-schema/json-schema-generic/src/main/scala/endpoints4s/generic/JsonSchemas.scala

```diff
@@ -109,7 +115,38 @@ trait JsonSchemas extends algebra.JsonSchemas {
             }(ht => (ht.head, ht.tail))
       }
 
-    implicit def consOptRecord[L <: Symbol, H, T <: HList, DH <: Option[docs], DT <: HList](implicit
+    implicit def consGenericRecord[
+        L <: Symbol,
+        H,
+        T <: HList,
+        DH <: Option[docs],
+        DT <: HList,
+        VH <: Option[H],
+        VT <: HList
+    ](implicit
+        labelHead: Witness.Aux[L],
+        jsonSchemaHead: JsonSchema[H],
+        jsonSchemaTail: DerivedGenericRecord[T, DT, VT]
+    ): DerivedGenericRecord[FieldType[L, H] :: T, DH :: DT, VH :: VT] =
+      new DerivedGenericRecord[FieldType[L, H] :: T, DH :: DT, VH :: VT] {
+        def record(docs: DH :: DT, defaultValues: VH :: VT) = {
+          val recordHead = defaultValues.head match {
+            case Some(defaultValue) =>
+              optFieldWithDefault[H](labelHead.value.name, defaultValue, docs.head.map(_.text))
+            case None =>
+              field[H](labelHead.value.name, docs.head.map(_.text))
+          }
```

Comment:
This is the logic that maps case class fields with a default value to an optional JSON field with default value, and regular case class fields to a mandatory JSON field.